bundle/deploy/terraform/load.go

```diff
@@ -57,6 +69,6 @@ func ValidateState(state *tfjson.State) error {
 	return nil
 }
 
-func Load() bundle.Mutator {
-	return &load{}
+func Load(modes ...loadMode) bundle.Mutator {
```

Comment:
No, infact there probably wont be multiple modes. The reason I went with this approach is this is more readable at the call site:
```
terraform.Load(terraform.NoErrorOnEmptyState)
```

The alternatives include a boolean field (not readable) or a enum for the "goal/mode" of load which also did non seem like a good fit here.

Reply:
Oh I see, it's for zero or one. We can change the callsite to be explicit ("ErrorOnEmptyState") or have two public methods, one with one parameter and one with no parameters. What do you think?

Reply:
Exactly it's zero to one. I'm against having two methods because there's no need for it. 

Between `ErrorOnEmptyState` and `NoErrorOnEmptyState`, I am ok with either. I prefered `NoErrorOnEmptyState` here because the branching is clearer in that case. Note, we cannot call `TerraformToBundle` if the state is empty. .

Reply:
As a general rule, don't use negatives in constant names because double negatives read odd. I.e. if I write `if !NoErrorOnFoo` is harder to parse than `if ErrorOnFoo`.

Reply:
You only need two Load methods, essentially two constructors.
```go
func Load() bundle.Mutator { ... } // or LoadDefault()
func LoadAllowEmptyState() bundle.Mutator { ... } // in which case you don't need a separate options enum
// or
func Load(allowEmptyState boolean) // or LoadWithOptions
```

The main validation logic could still be the same. This approach just seems unnecessarily generic for what we're trying to do here.

Reply:
We follow this pattern of variadic args for 0/1 modes in other places too. For example:
https://github.com/databricks/cli/blob/d70d7445c4f8f873959dd45ae68f8c42617d9353/libs/filer/filer.go#L17-L21

IMO it's better to expose a single public method here, and allow users to toggle the validation they require using the mode arg. Two public APIs seem unnecessary even if the most the code would be common. 

I have refactored the code to use `ErrorOnEmptyState` instead though now.

Reply:
Ok, happy to go with precedent if it exists. I guess this kind of pattern is used for functional options too. Given that this is also quasi-internal to DABs, maybe it is fine.

Reply:
I guess I'm a purist, but to me, it doesn't feel good to allow users to call functions in meaningless ways (in this case, with 2 or more values for this option). I'm pro-referential transparency, pro-"expose the exact interface you permit".